### ISSUE TITLE
Add sidebar slide effect

### ISSUE THREAD

**Issue by digitalethics** [johnfactotum/foliate#388]
Currently the sidebar mode just kind of "pops" in without any "slide" effect.

> I don't think this is currently possible with GTK. It is possible with `libdazzle`. I guess we can use `libdazzle`, but I'll want to make it optional, like the way we use `libhandy` at the moment; I don't want to hard-depend on a library just to add a few animations.

Yes, I think making it optional is a good idea. Do you know what [Setzer](https://github.com/cvfosammmm/Setzer) uses? _Peek_ didn't capture it very well but I think it shows the idea:

![setzer_sidebarslide](https://user-images.githubusercontent.com/57445091/82922505-f7794300-9f79-11ea-855b-feccdf0a5821.gif)

**Comment by johnfactotum** [comment]
Huh, interesting. Looking at the code, It seems that they are animating it manually rather than using GtkRevealer.

I'm a bit unconvinced, though. It's kind of a chore to write and maintain your own animation code, and also a lot of work to add a whole new library. I'm sure this will get into GTK at some point. Until then I think we can live without the animation.

**Comment by johnfactotum** [comment]
Not really. They're completely different. Maybe you can get it to work with some hack but I doubt it.

**Comment by johnfactotum** [comment]
It turns out this is quite easy to implement since GJS has Tweener built in. It's in the `sidebar-animation` branch now.

But adding the animation means the book is reflowed more than once when toggling the sidebar, which performs very badly. So it really needs to be much more sophisticated than simply animating the sidebar position. Otherwise the animation just makes everything look worse (it's basically the same issue as #459).

**Comment by digitalethics** [comment]
Ah, I see. Thanks so much for looking into strengthening the sidebar. Do you think it makes more sense to make the animation more sophisticated or first work on resolving #459?

**Comment by johnfactotum** [comment]
It's two manifestations of the same problem:
- Resizing = Bad
- More resizing = Extra bad

Solution is either
- Don't resize
- Hide the bad

**Comment by johnfactotum** [comment]
Currently libhandy is optional but in the future we will require libhandy/libadwaita (#612), and we will use libhandy/libadwaita's Flap widget, which has the sliding effect built in.

**Comment by johnfactotum** [comment]
Fixed in the [`gtk4`](https://github.com/johnfactotum/foliate/tree/gtk4) branch.
